source/server/admin/stats_handler.cc

```diff
@@ -71,78 +72,39 @@ Http::Code StatsHandler::handlerStatsRecentLookupsEnable(Http::ResponseHeaderMap
   return Http::Code::OK;
 }
 
-Admin::RequestPtr StatsHandler::makeRequest(AdminStream& admin_stream) {
+Admin::RequestPtr StatsHandler::makeRequest(AdminStream& admin_stream, bool is_prometheus_path) {
   StatsParams params;
   Buffer::OwnedImpl response;
   Http::Code code = params.parse(admin_stream.getRequestHeaders().getPathValue(), response);
   if (code != Http::Code::OK) {
     return Admin::makeStaticTextRequest(response, code);
   }
 
-  if (params.format_ == StatsFormat::Prometheus) {
-    // TODO(#16139): modify streaming algorithm to cover Prometheus.
-    //
-    // This may be easiest to accomplish by populating the set
-    // with tagExtractedName(), and allowing for vectors of
-    // stats as multiples will have the same tag-extracted names.
-    // Ideally we'd find a way to do this without slowing down
-    // the non-Prometheus implementations.
-    Buffer::OwnedImpl response;
-    prometheusFlushAndRender(params, response);
-    return Admin::makeStaticTextRequest(response, code);
-  }
-
   if (server_.statsConfig().flushOnAdmin()) {
     server_.flushStats();
   }
 
-  return makeRequest(server_.stats(), params,
-                     [this]() -> Admin::UrlHandler { return statsHandler(); });
-}
-
-Admin::RequestPtr StatsHandler::makeRequest(Stats::Store& stats, const StatsParams& params,
-                                            StatsRequest::UrlHandlerFn url_handler_fn) {
-  return std::make_unique<StatsRequest>(stats, params, url_handler_fn);
-}
-
-Http::Code StatsHandler::handlerPrometheusStats(Http::ResponseHeaderMap&,
-                                                Buffer::Instance& response,
-                                                AdminStream& admin_stream) {
-  return prometheusStats(admin_stream.getRequestHeaders().getPathValue(), response);
-}
-
-Http::Code StatsHandler::prometheusStats(absl::string_view path_and_query,
-                                         Buffer::Instance& response) {
-  StatsParams params;
-  Http::Code code = params.parse(path_and_query, response);
-  if (code != Http::Code::OK) {
-    return code;
-  }
-
-  if (server_.statsConfig().flushOnAdmin()) {
-    server_.flushStats();
+  if (params.format_ == StatsFormat::Prometheus || is_prometheus_path) {
```

Comment:
this seems a little awkward; can we change the call-site `/stats/prometheus` to set the format,  and drop the extra bool?